src/fmmax/fmm.py

```diff
@@ -319,6 +323,13 @@ def _incompatible(arr: jnp.ndarray, reference_shape: Tuple[int, ...]) -> bool:
                 f"shapes {self.omega_script_k_matrix.shape}  and {self.eigenvectors.shape}."
             )
 
+        if self.tangent_vector_field is not None:
```

Comment:
Good idea to add some validation here. Just two nits:
-no need to nest the if statements
-the error message could be improved. It mentions getting "shapes" but you are printing "ndim". In this case I would just print the shapes, as this will be even more helpful for debugging.

Reply:
Thank you for the comments! I made the changes accordingly. Will merge when the it passes the tests.